code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1445,3 +1446,69 @@ obj/machinery/computer/pandemic/proc/replicator_cooldown(var/waittime)
 		ui = new(user, src, ui_key, "chem_heater.tmpl", "ChemHeater", 350, 270)
 		ui.set_initial_data(data)
 		ui.open()
+
+///////////////////////////////////////////////////////////////////////////
+
+/obj/machinery/chem_dispenser/soda
+	name = "soda dispenser"
+	anchored = 1
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "soda_dispenser"
+	energy = 100
+	max_energy = 100
+	amount = 30
+	recharge_delay = 5
+	uiname = "Soda Dispenser"
+	dispensable_reagents = list("water","ice","coffee","cream","tea","icetea","cola","spacemountainwind","dr_gibb","space_up","tonic","sodawater","lemon_lime","sugar","orangejuice","limejuice","tomatojuice")
+
+/obj/machinery/chem_dispenser/soda/attackby(var/obj/item/O as obj, var/mob/user as mob)
+
+		if(default_unfasten_wrench(user, O))
+				return
+
+		if (istype(O,/obj/item/weapon/reagent_containers/glass) || \
+				istype(O,/obj/item/weapon/reagent_containers/food/drinks/drinkingglass) || \
+				istype(O,/obj/item/weapon/reagent_containers/food/drinks/shaker))
+
+				if (beaker)
+						return 1
+				else
+						src.beaker =  O
+						user.drop_item()
+						O.loc = src
+						update_icon()
+						src.updateUsrDialog()
+						return 0
+
+
+
+/obj/machinery/chem_dispenser/beer
+	name = "booze dispenser"
+	anchored = 1
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "booze_dispenser"
+	energy = 100
+	max_energy = 100
+	amount = 30
+	recharge_delay = 5
+	uiname = "Booze Dispenser"
+	dispensable_reagents = list("lemon_lime","sugar","orangejuice","limejuice","sodawater","tonic","beer","kahlua","whiskey","wine","vodka","gin","rum","tequilla","vermouth","cognac","ale")
+
+/obj/machinery/chem_dispenser/beer/attackby(var/obj/item/O as obj, var/mob/user as mob)
+
+		if(default_unfasten_wrench(user, O))
+				return
+
+		if (istype(O,/obj/item/weapon/reagent_containers/glass) || \
+				istype(O,/obj/item/weapon/reagent_containers/food/drinks/drinkingglass) || \
+				istype(O,/obj/item/weapon/reagent_containers/food/drinks/shaker))
+
+				if (beaker)
+						return 1
+				else
+						src.beaker =  O
+						user.drop_item()
+						O.loc = src
+						update_icon()
+						src.updateUsrDialog()
+						return 0
```

Comment:
no, usable prototypes create problems if you ever want to do type checks for something